sunpy/map/tests/test_map_factory.py

```diff
@@ -18,235 +18,226 @@
 import sunpy.data.test
 
 
-filepath = sunpy.data.test.rootdir
+filepath = pathlib.Path(sunpy.data.test.rootdir)
 a_list_of_many = [os.fspath(f) for f in pathlib.Path(filepath, "EIT").glob("*")]
 a_fname = a_list_of_many[0]
 
-AIA_171_IMAGE = os.path.join(filepath, 'aia_171_level1.fits')
-RHESSI_IMAGE = os.path.join(filepath, 'hsi_image_20101016_191218.fits')
-
-
-# ==============================================================================
-# Map Factory Tests
-# ==============================================================================
-class TestMap:
-    def test_mapsequence(self):
-        # Test making a MapSequence
-        sequence = sunpy.map.Map(a_list_of_many, sequence=True)
-        assert isinstance(sequence, sunpy.map.MapSequence)
-
-    def test_composite(self):
-        # Test making a CompositeMap
-        comp = sunpy.map.Map(AIA_171_IMAGE, RHESSI_IMAGE, composite=True)
-        assert isinstance(comp, sunpy.map.CompositeMap)
-
-    def test_patterns(self):
-        # Test different Map pattern matching
-
-        # File name
-        eitmap = sunpy.map.Map(a_fname)
-        assert isinstance(eitmap, sunpy.map.GenericMap)
-
-        # Directory
-        directory = pathlib.Path(filepath, "EIT")
+AIA_171_IMAGE = filepath / 'aia_171_level1.fits'
+RHESSI_IMAGE = filepath / 'hsi_image_20101016_191218.fits'
+
+amap = sunpy.map.Map(AIA_171_IMAGE)
+
+valid_map_inputs = [(a_fname, ),
+                    (pathlib.Path(a_fname), ),
+                    (filepath / "EIT", ),
+                    (os.fspath(filepath / "EIT"), ),
+                    (filepath / "EIT" / "*", ),
+                    (amap, ),
+                    (amap.data, amap.meta),
+                    ((amap.data, amap.meta), ),
+                    ]
+
+
+@pytest.mark.parametrize('args1', valid_map_inputs)
+@pytest.mark.parametrize('args2', valid_map_inputs)
+def test_two_map_inputs(args1, args2):
+    out = sunpy.map.Map(*args1, *args2)
+    if isinstance(out, list):
+        for m in out:
+            assert isinstance(m, sunpy.map.GenericMap)
+    else:
+        assert isinstance(out, sunpy.map.GenericMap)
+
+
+def test_mapsequence():
+    # Test making a MapSequence
+    sequence = sunpy.map.Map(a_list_of_many, sequence=True)
+    assert isinstance(sequence, sunpy.map.MapSequence)
+
+
+def test_composite():
+    # Test making a CompositeMap
+    comp = sunpy.map.Map(AIA_171_IMAGE, RHESSI_IMAGE, composite=True)
+    assert isinstance(comp, sunpy.map.CompositeMap)
+
+
+def test_patterns():
```

Comment:
it might be nice if these were individual tests, but that's *really* nitpicky :grinning:

Reply:
Yeah, I thought about splitting them up and making them nicer (maybe parametrizing them), but held off since it seemed to be non-trivial